### ISSUE TITLE
Dendrite sends read receipts regardless of user preference

### ISSUE THREAD

**Issue by necropola** [matrix-org/dendrite#3284]
### Background information
- dendrite-0.13.4+317b101
- postgressql-15.4
- Running native on ArchLinux with nginx as SSL proxy
- go-1.21.4
- element-web-v1.11.51

### Description
- Read Receipts are sent even when disabled in element-web User Preferences/Presence.
- Disabling inbound/outbound presence in dendrite config has no effect on receiving/sending receipts.  
- I'm getting the Read Receipts from dendrite on a different user/server (synapse with same element-web version)

### Steps to reproduce
- Disable sending Read Receipts in element-web and post a message to a chat room 
- Check on the other user/server with "show read receipts from other users" enabled that you are still getting read receipts
- Doing the same from the other user works, i. e. read receipts are not sent when disabled by user preference.

Basically synapse honors this user preferecen and dendrite ignores it.

**Comment by S7evinK** [comment]
Imo, not a Dendrite issue, more Element Web..
The client knows what to send and what not..

We can't check everything a random client sends us.

**Comment by S7evinK** [comment]
Right, Element Web does indeed **not** use `m.read.private` ([Spec](https://spec.matrix.org/latest/client-server-api/#receipts)) as the `receiptType` since we don't announce support for Matrix 1.4.

If you really want this _now_, you can patch 
https://github.com/matrix-org/dendrite/blob/fd11e65a9d113839177f9f7a32af328a0292b195/clientapi/routing/routing.go#L132-L153
to include 1.4

**Comment by necropola** [comment]
Let's see, if I understand this correctly:
- Clients when talking to Servers may only use feature/messages from a Matrix Protocol Version the Server has announced to support.
-  Servers may only announce support for a Matrix Protocol Version, if they fully support **ALL** features/messages of it.
-  Even though dendrite already supports the required feature/message (`m.read.private`), the clients (element) is not supposed to use/send it, because dendrite did not announce (full) support for the Matrix Protocol Version (`v1.4`) it belongs to.
- I could (wrongly) claim full support for `v1.4` by patching the dendrite sources (that's easy), but this would make clients assume my dendrite homeserver fully supports all features, e. g. Threads (`m.threadi`)  as per https://github.com/matrix-org/matrix-spec-proposals/pull/3440, https://github.com/matrix-org/matrix-spec-proposals/pull/3816, https://github.com/matrix-org/matrix-spec-proposals/pull/3856, and https://github.com/matrix-org/matrix-spec-proposals/pull/3715.

This makes it even harder for dendrite (server) developers to keep up with possible feature creep introduced by other (client and server) developers and gives a lot of power to the people deciding what goes in to a Matrix Protocol Version.

**Comment by S7evinK** [comment]
Exactly.

As you've seen, I've created several [issues](https://github.com/matrix-org/dendrite/issues?q=is%3Aissue+is%3Aopen+Declare+support+) regarding support for specific versions. Thing is, we already claim to have support for 1.1 and 1.2, which in reality we don't.

**Comment by necropola** [comment]
Announcing support for `v1.4` (and `v1.3`) by patching my dendrite sources does indeed *fix* this issue. Though this isn't of course a **proper fix**. What would be the best way to  go forward? Add all features (MSCS) to dendrite to fully support the matrix protocol up to version `v1.x` (with `x` being what?). 

- My golang skills are quite rusty and have never been particularly good.
- I'm kinda worried about the future of the non-mainstream matrix clients and servers. Seeing how element and synapse are pullling ahead.
- How does/will the new license and repository impact dendrite? (https://github.com/matrix-org/dendrite/discussions/3281l)

**Comment by spaetz** [comment]
> What would be the best way to go forward?
Indeed, the only way right now is to implement the missing features, to support matrix up to 1.4. there are meta issues for each version.

There are ideas floating to announce feature support instead of version support, but that is not what the spec says right now.
